src/Relude/Extra/Type.hs

```diff
@@ -114,3 +119,25 @@ Maybe (Snd (Int, Text)) :: *
 type family Snd (t :: k) :: k' where
     Snd '(_, y) = y
     Snd  (_, y) = y
+
+
+{- | Check that a value is an element of a list:
+
+>>> ok (Proxy :: Proxy (Elem Bool '[Int, Bool]))
+OK
+
+>>> ok (Proxy :: Proxy (Elem String '[Int, Bool]))
+...
+... [Char]...'[Int, Bool...
+...
+-}
+type Elem e es = ElemGo e es es
+
+-- 'orig' is used to store original list for better error messages
+type family ElemGo e es orig :: Constraint where
```

Comment:
I've been thinking about this type family for some type. And I came to conclusion that it's better to implement a type family with the following kind signature:

```haskell
type family Elem (e :: t) (es :: [t]) :: Bool
```

Sorry for derailing the issue into another direction... I find it a bit unsatisfying that you can't restore `Boo` behavior from a version that returns `Constraint`. But if you have a type family that returns `Bool` then you can easily write your own `IsElem` type family that returns either `()` or type error. So let's go with a simpler solution here.

Also, let's use `:kind!` command in tests here. Since we are not using type errors, `:kind!` is better to test such type families :slightly_smiling_face: